### PR TITLE
add and configure storybook

### LINKED ISSUE #18: Add Storybook
[Storybook](https://storybook.js.org/) allows us to build components in isolation from the rest of the app.

This will enable us to focus on building effective components _first_ before worrying about integrating them into the layout and greater structure of the app.

### PR DESCRIPTION
**Pull request by with-heart**
This PR closes #18 by adding and configuring [Storybook](https://storybook.js.org/) and related libraries and making a few changes to the codebase to support this change. 

This means we can now rely on Storybook to develop our components in isolation from the rest of the app.

---

To add Storybook, I followed the [Get started with Storybook and Next.js](https://storybook.js.org/blog/get-started-with-storybook-and-next-js/) guide.

Storybook configuration can be found in the `.storybook` directory. `.storybook/main.js` contains configuration for Storybook itself and `.storybook/preview.js` contains global configuration for stories and addons.

To make Storybook work with Chakra UI, I added [`@chakra-ui/storybook-addon`](https://chakra-ui.com/getting-started/with-storybook).

I'll explain the other changes in comments on the relevant code.

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { theme } from '../theme'
+
+// de-optimize Image component
+const OriginalNextImage = NextImage.default
+Object.defineProperty(NextImage, 'default', {
+  configurable: true,
+  value: (props) => <OriginalNextImage {...props} unoptimized />,
+})
+
+export const parameters = {
```
Comment (with-heart) [inline .storybook/preview.js:12]: [`parameters`](https://storybook.js.org/docs/react/writing-stories/parameters) define global configuration that can be accessed by Storybook addons or individual stories.

File: .storybook/preview.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  chakra: {
+    theme,
+  },
```
Comment (with-heart) [inline .storybook/preview.js:15]: `@chakra-ui/storybook-addon` reads options from `parameters.chakra`. We pass our `theme` here so our stories match our theme.

File: .storybook/preview.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+export const decorators = [
```
Comment (with-heart) [inline .storybook/preview.js:18]: [`decorators`](https://storybook.js.org/docs/react/writing-stories/decorators) are essentially wrappers around our stories. `decorators` defined in this file are global, meaning every story will be wrapped by them when rendered.

File: components/Header.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+import Header from './Header'
+
+export default {
+  component: Header,
+  title: 'Header',
+}
```
Comment (with-heart) [inline components/Header.stories.tsx:7]: Stories written in the [Component Story Format](https://storybook.js.org/docs/react/api/csf) require a `default` exported object that contains metadata about the stories in that file. At a minimum, it requires the `component` and `title` (which determines how the stories are listed in the Storybook sidebar).

**Comment by briangershon** [comment]
Thanks @with-heart!

I'd be happy to review this after we get the [MVP Milestone](https://github.com/Developer-DAO/school-of-code/milestone/1) out the door!

**Comment by briangershon** [comment]
I rebased this branch on latest `main` and resolved conflicts.

**Comment by briangershon** [comment]
One conflict was `logo.svg` which was in two places in `main` branch, neither used. Deleted both.

File: components/Header.tsx [hunk 5]
```diff
@@ -14,10 +13,12 @@ export default function Header() {
       justify="space-between"
     >
       <Image
-        src={MainHeaderLogo}
+        src="/logo.svg"
+        width={303}
+        height={60}
```
[the remark this answers is not in the document]
Comment (briangershon) [reply components/Header.tsx]: Logo was no longer used in `main` branch, removed.

**Review by briangershon — APPROVED** [review]
Thanks @with-heart!

- I rebased and resolve merge conflicts with `main`
- With refactor of `_app.tsx` I ran through and made sure site still works as it does on `main`.
- I verified Storybook runs. Upgraded dependencies.
- The example storybook file was `<Header />` but that's become a more complex component (depends on Next router) and will take more time to get that working, so I replaced it with `<SchoolOfCodeLogo />` instead.

Looks good-to-go!

**Pull request merged**